src/Picasso_MarchingCubes.hpp

```diff
@@ -0,0 +1,650 @@
+/****************************************************************************
+ * Copyright (c) 2021 by the Picasso authors                                *
+ * All rights reserved.                                                     *
+ *                                                                          *
+ * This file is part of the Picasso library. Picasso is distributed under a *
+ * BSD 3-clause license. For the licensing terms see the LICENSE file in    *
+ * the top-level directory.                                                 *
+ *                                                                          *
+ * SPDX-License-Identifier: BSD-3-Clause                                    *
+ ****************************************************************************/
+
+#ifndef PICASSO_MARCHINGCUBES_HPP
+#define PICASSO_MARCHINGCUBES_HPP
+
+#include <Cajita.hpp>
+
+#include <Kokkos_Core.hpp>
+
+#include <mpi.h>
+
+#include <fstream>
+#include <string>
+
+namespace Picasso
+{
+namespace MarchingCubes
+{
+//---------------------------------------------------------------------------//
+struct LookupTable
+{
+    // Facet counts.
+    inline static short counts[256] = {
```

Comment:
CI is warning about those `inline` variables needing c++17 - this also broke my Summit build (complaining that they need to be static even though they are).

Reply:
I switched these to `constexpr` variables instead. Hopefully that should work.

Reply:
The CI didn't like that so now I am doing the old-school static definition in a separate source file.

Reply:
Still having issues with these being undefined on the device

Reply:
@sslattery maybe we just make these `Kokkos::Array`s and improve it later?

Reply:
I just pushed this - we will see if CUDA likes it.